cypher-shell/src/main/java/org/neo4j/shell/prettyprint/TableOutputFormatter.java

```diff
@@ -92,9 +99,20 @@ private int formatResultAndCountRows(String[] columns,
         return columnSizes;
     }
 
+    /**
+     * The length of a column, where Numbers are always getting enough space to fit the highest number possible.
+     */
+    private int columnLengthForValue(Value value) {
+        if (value instanceof NumberValueAdapter ) {
+            return 19; // The number of digits of Long.Max
+        } else {
```

Comment:
I think it would be nice to avoid doing this when the result fits within `numSampleRows`. We could pass `records.hasNext()` as an additional parameter to `calculateColumnSizes` to decide if this should be applied or not.